hedera-node/src/main/java/com/hedera/services/store/contracts/HederaStackedWorldStateUpdater.java

```diff
@@ -268,6 +256,18 @@ private boolean isMissingTarget(final Address alias) {
         return !trackingAccounts().exists(accountIdFromEvmAddress(target));
     }
 
+    private int totalAllocatedIdsInTxn() {
+        var totalAllocatedIds = 0;
+        Optional<WorldUpdater> parentOptional = parentUpdater();
+        while (parentOptional.isPresent()) {
+            final var parent = (HederaWorldUpdater) parentOptional.get();
+            totalAllocatedIds += parent.numberOfIdsAllocatedByStacked();
+            parentOptional = parent.parentUpdater();
+        }
+
+        return totalAllocatedIds + numAllocatedIds;
+    }
+
```

Comment:
If we are concerned about the overhead, we can pass the total number of previously allocated ids in the constructor and save some cycles here.